components/post/Tags.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { X } from "lucide-react";
+
+const Tags = ({ setValue }: { setValue: any }) => {
+  const [tagInput, setTagInput] = useState("");
+  const [tags, setTags] = useState<string[]>([]);
+
+  useEffect(() => {
+    setValue("tags", tags);
```

Comment:
Needed to pull this into it's own useEffect, outside of the event handlers in the other two useEffects - there seemed to be some sort of race condition which wasn't adding the last item in the array